File: tests/js/lib/check.js
/* Tiny shared assertion harness -- no framework, matching this project's
 * own "stay dependency-free" choice. Each test file creates one Checker,
 * calls .check(label, cond) for every assertion, then .finish() to print
 * the summary and set process.exitCode. */
class Checker {
  constructor() {
    this.pass = 0;
    this.fail = 0;
  }
  check(label, cond) {
    if (cond) {
      this.pass++;
      console.log(`PASS: ${label}`);
    } else {
      this.fail++;
      console.log(`FAIL: ${label}`);
    }
  }
  finish() {
    console.log(`\n${this.pass} passed, ${this.fail} failed`);
    process.exitCode = this.fail > 0 ? 1 : 0;
  }
}

module.exports = { Checker };

File: tests/python/lib/check.py
"""Tiny shared assertion harness -- no framework (pytest/unittest), matching
this project's own dependency-free choice. Each test file creates one
Checker, calls .check(label, cond) for every assertion, then .finish()."""
import sys


class Checker:
    def __init__(self):
        self.passed = 0
        self.failed = 0

    def check(self, label, cond):
        if cond:
            self.passed += 1
            print(f"PASS: {label}")
        else:
            self.failed += 1
            print(f"FAIL: {label}")

    def finish(self):
        print(f"\n{self.passed} passed, {self.failed} failed")
        sys.exit(1 if self.failed else 0)
